onnxruntime/core/framework/execution_provider.cc

```diff
@@ -37,4 +40,99 @@ common::Status IExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>&
 }
 
 #endif
+
+ExecutionProviders::ExecutionProviders() {
+#ifdef _WIN32
+  // Register callback for ETW capture state (rundown)
+  etw_callback_key_ = "ExecutionProviders_rundown_";
```

Comment:
Why do you move the code here?
ExecutionProviders is not a singleton. It can have multiple instances.